cliquet/views/oauth.py

```diff
@@ -33,15 +37,15 @@ def persist_state(request):
     It will be compared when return from login page on OAuth server.
     """
     state = uuid.uuid4().hex
-    request.registry.session.set(state, request.validated['redirect'])
+    redirect_url = fxa_conf(request, 'webapp.redirect_url')
+    request.registry.session.set(state, redirect_url)
```

Comment:
Otherwise a potential vector attack is to generate a lot of states and then try brute forcing.